cmake/Subdirectories.cmake

```diff
@@ -29,6 +29,33 @@ function(add_engine_gem_json_external_subdirectories gem_path)
                 add_engine_gem_json_external_subdirectories(${real_external_subdir})
             endif()
         endforeach()
+
+        # see if this o3de object has any pal/restricted external subdir additions
+        set(pal_platforms ${PAL_PLATFORM_NAME})
+        list(APPEND pal_platforms ${LY_PAL_TOOLS_ENABLED})
+        foreach(platform ${pal_platforms})
+            string(TOLOWER ${platform} platform_lower)
+            o3de_restricted_path(${gem_json_path} gem_restricted_path gem_parent_relative_path)
+            if(gem_restricted_path)
+                o3de_pal_dir(pal_dir ${gem_path}/Platform/${platform} ${gem_restricted_path} ${gem_path} ${gem_parent_relative_path})
+                set(restricted_gem_json_path ${pal_dir}/gem_${platform_lower}.json)
+                if(EXISTS ${restricted_gem_json_path})
+                    read_json_external_subdirs(restricted_gem_external_subdirs ${restricted_gem_json_path})
+                    foreach(restricted_gem_external_subdir ${restricted_gem_external_subdirs})
+                        file(REAL_PATH ${restricted_gem_external_subdir} real_external_subdir BASE_DIRECTORY ${pal_dir})
+
+                        # Append external subdirectory if it is not in global property
+                        get_property(current_external_subdirs GLOBAL PROPERTY LY_EXTERNAL_SUBDIRS)
+                        if(NOT real_external_subdir IN_LIST current_external_subdirs)
+                            set_property(GLOBAL APPEND PROPERTY LY_EXTERNAL_SUBDIRS ${real_external_subdir})
+                            # Also append the project external subdirectores to the LY_EXTERNAL_SUBDIRS_ENGINE property
+                            set_property(GLOBAL APPEND PROPERTY LY_EXTERNAL_SUBDIRS_ENGINE ${real_external_subdir})
+                            add_engine_gem_json_external_subdirectories(${real_external_subdir})
+                        endif()
+                    endforeach()
+                endif()
+            endif()
+        endforeach()
```

Comment:
We should make this a function with Subdirectories.cmake to reduce the maintenance burden

```suggestion
    function(add_restricted_json_external_subdirectories gem_path external_subdirs_prop_name manifest_external_subdirs_prop_name)
        set(gem_json_path ${gem_path}/gem.json)
        # see if this o3de object has any pal/restricted external subdir additions
        set(pal_platforms ${PAL_PLATFORM_NAME})
        list(APPEND pal_platforms ${LY_PAL_TOOLS_ENABLED})
        foreach(platform ${pal_platforms})
            string(TOLOWER ${platform} platform_lower)
            o3de_restricted_path(${gem_json_path} gem_restricted_path gem_parent_relative_path)
            if(gem_restricted_path)
                o3de_pal_dir(pal_dir ${gem_path}/Platform/${platform} ${gem_restricted_path} ${gem_path} ${gem_parent_relative_path})
                set(restricted_gem_json_path ${pal_dir}/gem_${platform_lower}.json)
                if(EXISTS ${restricted_gem_json_path})
                    read_json_external_subdirs(restricted_gem_external_subdirs ${restricted_gem_json_path})
                    foreach(restricted_gem_external_subdir ${restricted_gem_external_subdirs})
                        file(REAL_PATH ${restricted_gem_external_subdir} real_external_subdir BASE_DIRECTORY ${pal_dir})

                        # Append external subdirectory if it is not in global property
                        get_property(current_external_subdirs GLOBAL PROPERTY LY_EXTERNAL_SUBDIRS)
                        if(NOT real_external_subdir IN_LIST current_external_subdirs)
                            set_property(GLOBAL APPEND PROPERTY ${external_subdirs_prop_name} ${real_external_subdir})
                            # Also append the project external subdirectores to the ${manifest_external_sbudir_prop_name} property
                            set_property(GLOBAL APPEND PROPERTY ${manifest_external_subdirs_prop_name} ${real_external_subdir})
                            add_restricted_json_external_subdirectories(${real_external_subdir} ${external_subdirs_prop_name} ${manifest_external_subdirs_prop_name})
                        endif()
                    endforeach()
                endif()
            endif()
        endforeach()
    endfunction()
```

Reply:
We would then just invoke 

```cmake
add_restricted_json_external_subdirectories(${real_external_subdir} LY_EXTERNAL_SUBDIRS LY_EXTERNAL_SUBDIRS_ENGINE)
```
 from in this function and have all the other `add_<manfiest>_gem_json_external_subdirectories` variants do the same.